### PR TITLE
Gql utility function get stats

### PR DESCRIPTION
**Pull request by asmab89**

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     *        ... on InterfaceObject1 {
+     *           field
+     *        }
+     *        ... on InterfaceObjectN {
+     *          field
+     *        }
+     *     }
+     */
+    const val INTERFACE_BASE_TAG = "#BASE#"
+    const val INTERFACE_TAG = "#INTERFACE#"
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:31]: what is the difference between `INTERFACE_BASE_TAG` and `INTERFACE_TAG`?

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const val SCALAR_TAG = "scalar"
+    const val OBJECT_TAG = "object"
+    const val UNION_STRING_TAG = "union"
+    const val INTERFACE_STRING_TAG = "interface"
+    const val ENUM_TAG = "enum"
+    const val LIST_TAG = "list"
+    const val INPUT_OBJECT_TAG = "input_object"
+    /**/
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:39]: why some `TAG` are in uppercase surrounded by `##` (eg `#BASE#`), whereas others are in lower case (eg `list`).
be consistent.
if those have completely different use in the code, then do not use the same `_TAG` suffix

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt [hunk 3]
```diff
@@ -7,12 +7,34 @@ import org.evomaster.core.problem.graphql.param.GQReturnParam
...
 import org.evomaster.core.search.gene.*
 import org.slf4j.LoggerFactory
+import java.util.*
 import javax.ws.rs.client.Entity
 
 object GraphQLUtils {
 
     private val log = LoggerFactory.getLogger(GraphQLUtils::class.java)
 
+    val history: Deque<String> = ArrayDeque<String>()
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt:17]: a public mutable static state? this needs to be refactored

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt (hunk 1 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:31]: eg:
let s take this schema:
```
interface Book {
  title: String!
}

type Textbook implements Book {
  title: String!
  courses: String!
}

type ColoringBook implements Book {
  title: String!
  colors: [String!]!
}

type Query {
  books: [Book!]!
}
```
And this Query:
```
{
  books {
    title
    ... on Textbook {
      courses # Only present in ColoringBook
    }
    ... on ColoringBook {
      colors # Only present in ColoringBook
    }
            }
}
```
INTERFACE_BASE_TAG: is used to annotate the interface type Book (so we can construct all the interface fields, like: title ) and
INTERFACE_TAG: is used to annotate the query field name books (which has the type: interface) (so we can construct ... on Textbook and ... on ColoringBook)

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt (hunk 2 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:39]: Indeed they are different. those surrounded by `## (eg #BASE#)`  are for a print function and the others `(in lower case(eg list))` are frequently used strings. 
Suffix changed.

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt (hunk 3 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt:17]: Did not understand this one.

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt (hunk 1 above)
Comment (arcuri82) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:31]: add this explanations as comments

Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GqlConst.kt:44]: Done

File: core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt (hunk 3 above)
Comment (asmab89) [reply core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt]: Already removed.

**Comment by asmab89** [comment]
Done

**Pull request merged**